gitcmd/gitcmd.go

```diff
@@ -30,6 +35,40 @@ type URLAuther interface {
 	// InsertAuth inserts authentication into the URL and returns it, or if the auther doesn't
 	// apply, returns the url without any modifications.
 	InsertAuth(url string) string
+	// InsertHTTPAuth inserts authentication into the request, if applicable.
+	InsertHTTPAuth(req *http.Request) error
```

Comment:
Missed comment about making a new interface for this? Actually, I think it should even belong in `githubutil`, not `gitcmd`, because this type of auth isn't used for local repositories.

Having the authers centralized in a single package while trying to make them multipurpose might be causing some difficulties. Might want to split the actual implementations out if that helps.